sympy/solvers/ode.py

```diff
@@ -2410,6 +2410,21 @@ def checkodesol(ode, sol, func=None, order='auto', solve_for_func=True):
     x = func.args[0]
     while s:
         if testnum == 0:
+            lhs, rhs = ode.lhs, ode.rhs
```

Comment:
It's probably better to just subtract these i.e. `ode_expr = ode.lhs - ode.rhs` then you only have one object to call subs with etc.

Reply:
Okk I will try it.

Reply:
I had done it. Please check.